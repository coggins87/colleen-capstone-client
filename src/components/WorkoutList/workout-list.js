import React from 'react'
import Workout from '../Workout/workout'
import uuid from 'uuid'
import './workout-list.css'
class WorkoutList extends React.Component {

render(){
  return (<>
    <h3 className="subheader">Complete as many reps as possible:</h3>
    {this.props.workouts.map(workout => { 
      return (
  <Workout key={uuid()} workout={workout}/>)

    })}
    </>)
}
}

export default WorkoutList