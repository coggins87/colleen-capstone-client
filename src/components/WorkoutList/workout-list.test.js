import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import WorkoutList from "./workout-list";

jest.mock("uuid", () => jest.fn(() => "1"));
describe(`WorkoutList component`, () => {
  const props = {
    workouts: [
      {
        workout_length: 1,
        movements: [
          { movement_name: "name" },
          { reps: "reps" },
          { equipment: "equip" }
        ]
      }
    ]
  };

  it("renders the Workout given props", () => {
    const wrapper = shallow(<WorkoutList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
