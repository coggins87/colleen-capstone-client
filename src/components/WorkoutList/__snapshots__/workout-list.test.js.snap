// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutList component renders the Workout given props 1`] = `
<Workout
  key="1"
  workout={
    Object {
      "movements": Array [
        Object {
          "movement_name": "name",
        },
        Object {
          "reps": "reps",
        },
        Object {
          "equipment": "equip",
        },
      ],
      "workout_length": 1,
    }
  }
/>
`;
